### PR TITLE
nto pkg: performanceprofile - new methods added, fixes and unit test added

### PR DESCRIPTION
**Pull request by elenagerman**
new methods: 
- WithNodeSelector
- WithGloballyDisableIrqLoadBalancing

### CODE REVIEW

File: pkg/nto/performanceprofile_test.go [hunk 1]
```diff
@@ -0,0 +1,662 @@
...
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedError:          "",
+		},
+		{
+			testPerformanceProfile: buildInValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedError:          "PerformanceProfile's name is empty",
+		},
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:          "resourceVersion can not be set for Create requests",
```
Comment (kononovn) [inline pkg/nto/performanceprofile_test.go:238]: This test case should't return error. It has to create object successfully. You just need to remove resourceVersion from structure

File: pkg/nto/performanceprofile_test.go [hunk 2]
```diff
@@ -0,0 +1,662 @@
...
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedError:          nil,
+		},
+		{
+			testPerformanceProfile: buildInValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedError:          fmt.Errorf("PerformanceProfile's name is empty"),
+		},
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:          fmt.Errorf("PerformanceProfile cannot be deleted because it does not exist"),
```
Comment (kononovn) [inline pkg/nto/performanceprofile_test.go:269]: Please change Delete method. 
We have to return nil and set Object to nil if actual object doesn't exist on cluster.

File: pkg/nto/performanceprofile_test.go [hunk 3]
```diff
@@ -0,0 +1,662 @@
...
+			assert.Nil(t, err)
+		}
+	}
+}
+
+func TestPerformanceProfileWithHugePages(t *testing.T) {
+	testCases := []struct {
+		testHugePageSize  string
+		testPages         []v2.HugePage
+		expectedError     bool
```
Comment (kononovn) [inline pkg/nto/performanceprofile_test.go:330]: You can remove this var and you
if expectedErrorText != "" instead.

File: pkg/nto/performanceprofile_test.go [hunk 4]
```diff
@@ -0,0 +1,662 @@
...
+				*result.Definition.Spec.HugePages.DefaultHugePagesSize)
+			assert.Equal(t, testCase.testPages, result.Definition.Spec.HugePages.Pages)
+		}
+	}
+}
+
+func TestPerformanceProfileWithMachineConfigPoolSelector(t *testing.T) {
+	testCases := []struct {
+		testMachineConfigPoolSelector map[string]string
+		expectedError                 bool
```
Comment (kononovn) [inline pkg/nto/performanceprofile_test.go:386]: Please remove this var. Same as above.

File: pkg/nto/performanceprofile_test.go [hunk 5]
```diff
@@ -0,0 +1,662 @@
...
+			assert.NotNil(t, result)
+			assert.Equal(t, testCase.testMachineConfigPoolSelector, result.Definition.Spec.MachineConfigPoolSelector)
+		}
+	}
+}
+
+func TestPerformanceProfileWithNodeSelector(t *testing.T) {
+	testCases := []struct {
+		testNodeSelector  map[string]string
+		expectedError     bool
```
Comment (kononovn) [inline pkg/nto/performanceprofile_test.go:420]: Please remove this var same as above.

File: pkg/nto/performanceprofile_test.go [hunk 6]
```diff
@@ -0,0 +1,662 @@
...
+			assert.NotNil(t, result)
+			assert.Equal(t, testCase.testNodeSelector, result.Definition.Spec.NodeSelector)
+		}
+	}
+}
+
+func TestPerformanceProfileWithNumaTopology(t *testing.T) {
+	testCases := []struct {
+		testNumaTopology  string
+		expectedError     bool
```
Comment (kononovn) [inline pkg/nto/performanceprofile_test.go:454]: Please remove this var.

File: pkg/nto/performanceprofile_test.go [hunk 7]
```diff
@@ -0,0 +1,662 @@
...
+		} else {
+			assert.NotNil(t, result)
+			assert.Equal(t, testCase.testNumaTopology, *result.Definition.Spec.NUMA.TopologyPolicy)
+		}
+	}
+}
+
+func TestPerformanceProfileWithRTKernel(t *testing.T) {
+	testCases := []struct {
+		expectedError     bool
```
Comment (kononovn) [inline pkg/nto/performanceprofile_test.go:503]: Please remove this var.

File: pkg/nto/performanceprofile_test.go [hunk 8]
```diff
@@ -0,0 +1,662 @@
...
+		}
+	}
+}
+
+func TestPerformanceProfileWithWorkloadHints(t *testing.T) {
+	testCases := []struct {
+		rtHint              bool
+		perPodPowerMgmtHint bool
+		highPowerHint       bool
+		expectedError       bool
```
Comment (kononovn) [inline pkg/nto/performanceprofile_test.go:560]: Please remove this var.

**Review by kononovn — CHANGES_REQUESTED** [review]
Please re-base on top of the latest main.

File: pkg/nto/performanceprofile.go [hunk 9]
```diff
@@ -176,6 +200,30 @@ func (builder *Builder) WithMachineConfigPoolSelector(machineConfigPoolSelector
...
+	}
+
+	if len(nodeSelector) == 0 {
+		glog.V(100).Infof("'nodeSelector' argument cannot be empty")
+
+		builder.errorMsg = "'nodeSelector' argument cannot be empty"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
```
Comment (yprokule) [inline pkg/nto/performanceprofile.go:219]: ditto

**Review by kononovn — DISMISSED** [review]
/lgtm

File: pkg/nto/performanceprofile.go [hunk 10]
```diff
@@ -185,6 +229,16 @@ func (builder *Builder) WithNumaTopology(topologyPolicy string) *Builder {
...
 	}
 
+	if topologyPolicy == "" {
+		glog.V(100).Infof("The topologyPolicy is empty")
+
+		builder.errorMsg = "'topologyPolicy' argument cannot be empty"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
```
Comment (yprokule) [inline pkg/nto/performanceprofile.go:239]: please move it to the _topologyPolicy_ assertion clause above

File: pkg/nto/performanceprofile_test.go [hunk 11]
```diff
@@ -0,0 +1,606 @@
...
+			expectedErrorText: "'hugePageSize' argument is not in allowed list: [2M 1G]",
+		},
+		{
+			testHugePageSize:  "",
+			testPages:         defaultHugepages,
+			expectedErrorText: "'hugePageSize' argument cannot be empty",
+		},
+		{
+			testHugePageSize:  defaultHugepageSize,
+			testPages:         []v2.HugePage{},
```
Comment (yprokule) [inline pkg/nto/performanceprofile_test.go:350]: I'd consider adding 2 more tests:
* explicitly asking for 2 different NUMA nodes
```
        defaultHugePagesNodeOne = int32(0)
        defaultHugePagesNodeTwo = int32(1)
	defaultHugepages              = []v2.HugePage{
 		{
 			Size:  "2M",
 			Count: 32768,
 			Node:  &defaultHugePagesNodeOne,
 		},
                {
 			Size:  "2M",
 			Count: 32768,
 			Node:  &defaultHugePagesNodeTwo,
 		},
 	}
```
* not asking for a NUMA node
```
	defaultHugepages              = []v2.HugePage{
 		{
 			Size:  "2M",
 			Count: 32768,
 		},
        }
```

**Review by cdvultur — DISMISSED** [review]
/lgtm

**Pull request merged**